### PR TITLE
Amendments to DG and more test cases

### LINKED ISSUE #145: JavaDocs documentation for codebase
To add JavaDocs documentation for methods that are non-trivial and are currently not documented well.

### LINKED ISSUE #162: UG: The month command allows you to switch to the month view from any other view.
Switching to Month View: month

The month command allows you to switch to the month view from any other view.

### LINKED ISSUE #166: UG: Add dash into this explanation on storage stuff
Notes to users

    Commas, and pipes| are used in our application as command delimiters as well as for file storage purposes. Please avoid using these in your task descriptions as it could lead to parsing errors or storage issues.

    You have been informed!

### LINKED ISSUE #169: UG: delete, mark, priority, quit - to change to screenshots instead of text examples

### LINKED ISSUE #170: Rename save file in code to follow UG

### LINKED ISSUE #173: DG: Architecture adding components
add main, command parser, log

### LINKED ISSUE #181: DG: Saving Tasks to File to wrong file name
to clinton  .. instead

### LINKED ISSUE #182: DG: Add exceptions description for methods with exceptions

### LINKED ISSUE #183: DG: changing save file name in Storage component
to clintonData.txt

### LINKED ISSUE #184: DG: task saving format has changed
from <date>|<taskType>|<taskDescription>|<additionalData>

to 2024-04-15|T|O|M|write a report

### LINKED ISSUE #185: DG: marking Exporting .ics File Component as a future dev

### LINKED ISSUE #186: DG: adding more non-functional requirements
1. A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
2. The app is free for all users.
3. Should be able to work on a typical CLI and not require any third-party software.
4. Should response no more than 5 seconds for any user interactions.

Rephrase these

### PR DESCRIPTION
**Pull request by kyhjonathan**
Progress:
resolved #186,
resolved  #185, 
resolved #184, 
resolved #183, 
resolved #182, 
resolved #181, 
resolved #173, 
resolved #170.
resolved #169 
Remaining issues assigned under me are under development and will be pushed asap.

Update:
resolved #166 
resolved #162 
resolved #145

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -465,26 +478,34 @@ public void addTasksFromFile(Map<LocalDate, List<Task>> tasksFromFile) throws Ta
...
+* Handles exception if `./save/clintonData.txt` is in corrupted format.
+
+Below is a UML class diagram of the Storage component:
+![img.png](images/class/Storage.png)
 
 The `saveTasksToFile` method writes the tasks in a HashMap to the file in the following format:
 
 ```
-<date>|<taskType>|<taskDescription>|<additionalData>
+<date>|<taskType>|<markedStatus>|<priorityLevel>|<taskDescription>|<taskDetails>
```
Comment (NicholasTanYY) [inline docs/DeveloperGuide.md:492]: Great job updating the structure of the save file in the DG!

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -465,26 +478,34 @@ public void addTasksFromFile(Map<LocalDate, List<Task>> tasksFromFile) throws Ta
 **API** : [Storage.java](https://github.com/AY2324S2-CS2113-W13-2/tp/blob/master/src/main/java/storage/Storage.java)
 
 The 'storage' component:
-* Reads tasks from the formatted `./save/tasks.txt` file and appends to task hashmap.
-* Identifies unique tasks stored in task hashmap, parses and writes to `./save/tasks.txt` file
-* Handles exception if `./save/tasks.txt` is in corrupted format
+* Creates new file if `./save/clintonData.txt` does not exist.
+* Reads tasks from the formatted `./save/clintonData.txt` file and appends to task hashmap.
+* Identifies unique tasks stored in task hashmap, parses and writes to `./save/clintonData.txt` file.
+* Handles exception if `./save/clintonData.txt` is in corrupted format.
```
Comment (NicholasTanYY) [inline docs/DeveloperGuide.md:484]: Thanks for changing the name of the save file in the DG!

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -652,6 +673,10 @@ quickly create entries in their calendar.
 ## Non-Functional Requirements
 
 1. Should work on Linux, macOS, Unix and Windows as long as Java `11` is installed.
+2. A user with with an average typing speed should be able to add a task faster using commands than a mouse.
+3. The application will be free for all users
+4. The application should be able to work on typical CLI interface without the need of third-party software
+5. The application should be able to repond in no more than 5 seconds for any command
```
Comment (NicholasTanYY) [inline docs/DeveloperGuide.md:679]: Good points for NFR!

File: docs/UserGuide.md [hunk 4]
```diff
@@ -107,7 +107,7 @@ The application will move to the previous month view, displaying the month prior
 
 ### Switching to Month View: `month`
 
-The `month` command allows you to switch to the month view from any other view.
+The `month` command allows you to switch to the month view from week view.
```
Comment (NicholasTanYY) [inline docs/UserGuide.md:110]: This is clearer now!

**Review by NicholasTanYY — APPROVED** [review]
LGTM. Thanks for adding more test cases and more UML diagrams to make the structure of our programs clearer!

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -41,15 +41,19 @@ ical4J Library: [https://www.ical4j.org/](https://www.ical4j.org/)
 
 
 ## Architecture
-The Calendar application is designed with a modular architecture, consisting of the following components:
+The ***Architecture diagram*** given below shows the high-level architecture of CLI-nton.
+The application is designed with a modular architecture, consisting of the following components:
 
+![img.png](diagrams/architecture/architeturediagram.png)
```
Comment (Mahesh1772) [inline docs/DeveloperGuide.md:47]: Thanks for adding in the architecture diagram!

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -357,6 +361,8 @@ public static void addTask(LocalDate date, String taskDescription, TaskType task
 ### `addManager` Method
 
 The `addManager` method facilitates the management of adding tasks from user input along with the specified date. It prompts users to input task details, validates the input, and delegates the task creation process to the `addTask` method.
+The sequence diagram below illustrates the flow of the `addManager` method:
```
Comment (Mahesh1772) [inline docs/DeveloperGuide.md:364]: Good job on adding the addManager sequence!

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -387,11 +393,18 @@ Within the `addTask` method, a switch statement delineates the type of task bein
 
 ### Error Handling
 
-The implementation incorporates error handling mechanisms to effectively manage scenarios involving invalid inputs or unsupported task types. Exceptions such as `TaskManagerException` are employed to convey descriptive error messages, ensuring user guidance and application integrity.
+The implementation incorporates error handling mechanisms to effectively manage scenarios involving invalid inputs or unsupported task types. 
+Below is a list of exceptions and a description of how they are handled:
+* `TaskManagerException`: Handles errors that occur during task creation, such as invalid dates or task types.
```
Comment (Mahesh1772) [inline docs/DeveloperGuide.md:398]: Nice work listing out the Exceptions

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -465,26 +478,34 @@ public void addTasksFromFile(Map<LocalDate, List<Task>> tasksFromFile) throws Ta
...
+<date>|<taskType>|<markedStatus>|<priorityLevel>|<taskDescription>|<taskDetails>
 ```
 
 For example:
 
 ```
-2023-06-01|TODO|Buy groceries
-2023-06-02|EVENT|Meeting|2023-06-02|2023-06-03|09:00|11:00
-2023-06-05|DEADLINE|Submit report|2023-06-10|23:59
+2024-04-15|D|O|L|task 2|19/04/2024|1400
```
Comment (Mahesh1772) [inline docs/DeveloperGuide.md:498]: Thanks for the update of the storage format!

File: docs/UserGuide.md [hunk 9]
```diff
@@ -127,7 +127,7 @@ The application will switch to the month view from week view and current month w
 
 ### Switching to Week View: `week`
 
-The `week` command allows you to switch to the week view from any other view.
+The `week` command allows you to switch to the week view from month view.
```
Comment (Mahesh1772) [inline docs/UserGuide.md:130]: Good job on fixing the typo

**Review by Mahesh1772 — COMMENTED** [review]
LGTM!

File: src/test/java/storage/StorageTest.java [hunk 10]
```diff
@@ -1,9 +1,14 @@
...
 import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
 
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+
 class StorageTest {
 
     private static final Path TEST_FILE_PATH = Path.of("save", "test_tasks.txt");
```
Comment (clarencepohh) [inline src/test/java/storage/StorageTest.java:14]: This should be updated to clintonData.txt

File: src/test/java/storage/StorageTest.java [hunk 11]
```diff
@@ -18,4 +23,20 @@ void tearDown() throws IOException {
...
+        Path nonExistentPath = Path.of("nonexistent/directory/tasks.txt");
+
+        // Act and Assert
+        assertThrows(IOException.class, () -> Files.write(nonExistentPath, "test data".getBytes()));
+    }
+
+    @Test
+    void readFromFile_fileDoesNotExist_throwsIOException() {
+        Path nonExistentPath = Path.of("nonexistent/directory/tasks.txt");
+        assertThrows(IOException.class, () -> Files.readString(nonExistentPath));
```
Comment (clarencepohh) [inline src/test/java/storage/StorageTest.java:39]: Save file not updated here as well. Unless this is intended?

**Review by clarencepohh — CHANGES_REQUESTED** [review]
There are some errors that need to be fixed. Good job overall though!

**Comment by NicholasTanYY** [comment]
Great job resolving the merge conflicts!

File: src/test/java/storage/StorageTest.java (hunk 10 above)
Comment (clarencepohh) [reply src/test/java/storage/StorageTest.java:14]: On review, it is correct. My apologies

**Pull request merged**